src/main/java/org/mskcc/cbio/oncokb/web/rest/AccountResource.java

```diff
@@ -364,7 +365,10 @@ public Token createToken() {
     public void deleteToken(@RequestBody Token token) throws AuthenticationException {
         Optional<String> userLogin = SecurityUtils.getCurrentUserLogin();
         if (userLogin.isPresent() && token.getUser() != null) {
-            if (token.getUser().getLogin().equalsIgnoreCase(userLogin.get())) {
+            Optional<User> user = userService.getUserByLogin(userLogin.get());
+            boolean userIsAdmin = user.isPresent() && user.get().getAuthorities().stream().filter(authority -> authority.getName().equalsIgnoreCase(AuthoritiesConstants.ADMIN)).count() > 0;
```

Comment:
Should we make this a part of userservice? Maybe `isAdmin(String userLogin)`?

Reply:
Yeah for sure, could be a nice utility function to have